example/example.c

```diff
@@ -173,7 +173,8 @@ int main(int argc, char* argv[])
 
 	PDI_expose("mpi_rank", &pcoord_1d, PDI_OUT);
 	PDI_expose("mpi_size", &psize_1d, PDI_OUT);
-	PDI_event("init");
+	int ii = 0;
+	PDI_expose("iter", &ii, PDI_OUT);
```

Comment:
Why is this required?

Reply:
I need to expose `iter` before the init event is triggered or there's an error: `[MPI 000001][PDI] *** error: Error while triggering event `main_loop': while referencing `iter': Right_error: Unable to grant access for value reference`

Reply:
indeed, in the `.yml`, you set 
```yaml
deisa_arrays:
  global_t:
    start: [ $iter, '($dsize[0]-2)*$pcoord[0]', '($dsize[1]-2)*$pcoord[1]' ]
```
so then, when this is evaluated at initalization, on the `main_loop` event, on line 161:
```C++
Datatype_sptr type_sptr = type_tpl->evaluate(context());
```
`iter` is evaluated.

The issue, I believe is that the the `$timedim` element of the `start` array is not used in python at this point. So this could be ignored, and the evaluation is pointless, but it is still evaluated.

Can you add an issue about supporting partial evaluation of types, documenting this use-case and then, I guess you will indeed have to expose this just before the `main_loop` event.